### PR TITLE
unify services settings

### PR DESCRIPTION
**Pull request by wy65701436**

### CODE REVIEW

File: templates/nginx/service.yaml [hunk 1]
```diff
@@ -29,7 +33,14 @@ spec:
...
   name: {{ $nodePort.name }}
   labels:
+{{- if .Values.expose.nodePort.labels }}
+{{ toYaml $nodePort.labels | indent 4 }}
+{{- else }}
 {{ include "harbor.labels" . | indent 4 }}
+{{- end }}
+{{- with $nodePort.annotations }}
+  annotations:
+  {{- toYaml . | nindent 4 }}
```
Comment (zyyw) [inline templates/nginx/service.yaml:43]: maybe need a `{{- end }}`

File: templates/nginx/service.yaml [hunk 2]
```diff
@@ -6,7 +6,11 @@ metadata:
 {{- $clusterIP := .Values.expose.clusterIP }}
   name: {{ $clusterIP.name }}
   labels:
+{{- if .Values.expose.clusterIP.labels }}
+{{ toYaml $clusterIP.labels | indent 4 }}
+{{- else }}
```
Comment (ywk253100) [inline templates/nginx/service.yaml:11]: Should append the customized labels after the `harbor.labels`

**Review by ywk253100 — APPROVED** [review]

File: values.yaml [hunk 3]
```diff
@@ -53,13 +53,15 @@ expose:
...
-    annotations: {}
     ports:
       # The service port Harbor listens on when serving HTTP
       httpPort: 80
       # The service port Harbor listens on when serving HTTPS
       httpsPort: 443
+    # Annotations on the ClusterIP service
+    annotations: {}
+    # ClusterIP-specific labels
+    labels: {}
```
Comment (zyyw) [inline values.yaml:64]: please also update README.md for this change

File: values.yaml [hunk 4]
```diff
@@ -74,6 +76,10 @@ expose:
         port: 443
         # The node port Harbor listens on when serving HTTPS
         nodePort: 30003
+    # Annotations on the nodePort service
+    annotations: {}
+    # nodePort-specific labels
+    labels: {}
```
Comment (zyyw) [inline values.yaml:82]: please also update README.md for this change

File: values.yaml [hunk 5]
```diff
@@ -84,7 +90,10 @@ expose:
       httpPort: 80
       # The service port Harbor listens on when serving HTTPS
       httpsPort: 443
+    # Annotations on the loadBalancer service
     annotations: {}
+    # loadBalancer-specific labels
+    labels: {}
```
Comment (zyyw) [inline values.yaml:96]: please also update README.md for this change

File: README.md [hunk 6]
```diff
@@ -75,332 +75,336 @@ helm uninstall my-release
...
+| `expose.loadBalancer.annotations`                                    | The annotations used commonly for loadBalancer                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |                                       |
```
Comment (zyyw) [inline README.md:108]: line 108 is the same config of line 106? one should be `expose.loadBalancer.annotations`, and the other should be `expose.loadBalancer.labels`, I suppose.

File: README.md [hunk 7]
```diff
@@ -75,332 +75,336 @@ helm uninstall my-release
...
+| `expose.clusterIP.annotations`                                       | The annotations attached to the ClusterIP service                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | {}                                    |
```
Comment (zyyw) [inline README.md:92]: missing `expose.clusterIP.labels`

**Pull request merged**